### DISCUSSION TITLE
Config-driven scraping/testing with rod (towards AI/LLM-assisted automation)

### DISCUSSION THREAD

**Discussion by iljya** [go-rod/rod#1148] [General]
Hi everyone,

I'm wondering if there have been any attempts to create a config-language to drive scraping/testing automations. That is, rather than writing golang code to drive the browser, actions (click, input text, etc.) are configured in a config file. I am interested in this from the perspective of one-day having an LLM create these config files based on a high-level description of the task (and a webpage side-input).

I found this project which is exactly what I am looking for, but it appears to be abandoned:
https://github.com/go-rod/wayang

Many thanks for your attention!

**Comment by ysmood** [comment]
I think it's a valuable direction, but I don't have enough resource to work on it.

**Reply by iljya** [reply]
Thanks for the encouragement :)
